src/signalhandler.cpp

```diff
@@ -0,0 +1,146 @@
+/******************************************************************************\
```

Comment:
General comment to this file: It would be good if you could add some references from where you have all the information how to handle the signals. I guess you have used a lot of Qt help pages. Maybe other sources. It would be beneficial if you could add all these links in this file.

Reply:
Very, very annoyingly, I've actually lost my main source.

I went to check up on it again and simply couldn't find it.  It took some tracking down originally.  Most of it comes from fragments on Sourceforge.  Windows-only and unix-only examples abound - but I found an example that did both.  Then lost it - I actually got all the main coding done in an afternoon.  Came back, cleaned everything up, went to add references...  And couldn't find it.  Checked again before raising the pull request.  Still couldn't.

Basically, it's the same solution as in many threads across Sourceforge: Windows and Unix work differently.  The one "key" bit is the metadata trigger of the Qt call back.  I'll do another round of searches.

Reply:
Ah, found it.  My memory is useless.  I'd actually put it somewhere for safe keeping.  I should have also written that down somewhere I couldn't miss it.

My "main" references - as it brought together several commonly available sets of ideas, was `QCtrlSignals` (https://github.com/Skycoder42/QCtrlSignals.git).  It's BSD 3-clause.

Personally, I think my limited use would count as "fair use" but I'm not prepared to push it - I'd rather give the acknowledgement.  It simplified my life.  I then took that code apart as it had too much complexity.  (Look at all those files!)